plugin-server/src/main/utils.ts

```diff
@@ -35,3 +42,10 @@ export async function runInstrumentedFunction<T, E>({
         clearTimeout(timeout)
     }
 }
+
+const instrumentedFunctionDuration = new Histogram({
```

Comment:
Note that as it stands this calls to this will not be picked up from subprocesses i.e. Piscina workers. That's 2 of the 3 current call locations.

You'd need to add subprocess support [here](https://github.com/PostHog/posthog/blob/master/plugin-server/src/main/services/http-server.ts), or remove Piscina. I would be in favour of the latter, we're only using 1 worker thread as it stands so that doesn't appear to be a bottleneck, but it's probably a more scary task!

Reply:
Ah I totally didn't read your PR description 😕